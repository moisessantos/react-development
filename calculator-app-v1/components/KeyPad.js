import React, { memo } from 'react';
import PropTypes from 'prop-types';

import { KeyRow } from '.';

const KeyPad = ({ handleClick, ...otherProps }) => (
  <div className="button" {...otherProps}>
    <KeyRow buttons={['(', 'CE', ')', 'C']} handleClick={handleClick} />
    <KeyRow buttons={['1', '2', '3', '+']} handleClick={handleClick} />
    <KeyRow buttons={['4', '5', '6', '-']} handleClick={handleClick} />
    <KeyRow buttons={['7', '8', '9', '*']} handleClick={handleClick} />
    <KeyRow buttons={['.', '0', '=', '/']} handleClick={handleClick} />
  </div>
);

KeyPad.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

export default memo(KeyPad);
