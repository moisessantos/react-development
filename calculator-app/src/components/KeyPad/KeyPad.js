import React, { memo } from 'react';
import PropTypes from 'prop-types';

import { KeyRow } from '../';

const KeyPad = ({ handleClick, ...otherProps }) => (
  <div className="button" {...otherProps}>
    <KeyRow values={['(', 'CE', ')', 'C']} handleClick={handleClick} />
    <KeyRow values={['1', '2', '3', '+']} handleClick={handleClick} />
    <KeyRow values={['4', '5', '6', '-']} handleClick={handleClick} />
    <KeyRow values={['7', '8', '9', '*']} handleClick={handleClick} />
    <KeyRow values={['.', '0', '=', '/']} handleClick={handleClick} />
  </div>
);

KeyPad.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

export default memo(KeyPad);
