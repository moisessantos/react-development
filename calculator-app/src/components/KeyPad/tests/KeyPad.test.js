import React from 'react';
import { shallow } from 'enzyme';
import KeyPad from '../';

describe('KeyPad tests', () => {
    it('should render as expected', () => {
        const wrapper = shallow(<KeyPad handleClick={ jest.fn() } />)
        expect(wrapper).toMatchSnapshot();
    });
})