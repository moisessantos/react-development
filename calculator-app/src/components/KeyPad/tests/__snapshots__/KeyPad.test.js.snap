// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyPad tests should render as expected 1`] = `
<div
  className="button"
>
  <Memo(KeyRow)
    handleClick={[MockFunction]}
    values={
      Array [
        "(",
        "CE",
        ")",
        "C",
      ]
    }
  />
  <Memo(KeyRow)
    handleClick={[MockFunction]}
    values={
      Array [
        "1",
        "2",
        "3",
        "+",
      ]
    }
  />
  <Memo(KeyRow)
    handleClick={[MockFunction]}
    values={
      Array [
        "4",
        "5",
        "6",
        "-",
      ]
    }
  />
  <Memo(KeyRow)
    handleClick={[MockFunction]}
    values={
      Array [
        "7",
        "8",
        "9",
        "*",
      ]
    }
  />
  <Memo(KeyRow)
    handleClick={[MockFunction]}
    values={
      Array [
        ".",
        "0",
        "=",
        "/",
      ]
    }
  />
</div>
`;
